File: C#/AbstractType/Student.cs

namespace CShard
{
    public class Student
    {
        string name;
        int age;

        public Student(string name, int age)
        {
            this.name = name;
            this.age = age;
        }

        public void sayHello()
        {
            Console.WriteLine($"Hello, my name is {name}.\nI'm {age} years old.");
        }
    }
}

File: C++/AbstractType/Student.cpp

#include <string>
#include <cstdint>
#include <iostream>

class Student
{
public:
    std::string name;
    uint8_t age;

    void sayHello() {
        std::cout << "Hello, my name is " << name << ".\nI'm " << std::to_string(age) << " years old.";
    }

    Student(std::string _name, uint8_t _age)
    {
        name = _name;
        age = _age;
    }
    
};